### PR TITLE
Build: Tidy and improve cleanup rules

### PR DESCRIPTION
**Pull request by patrickbkr**
I'm unsure about `gen/nqp-version`. It feels too prominent to have simply been missed all this time. Is there a reason to keep it around even on `realclean`? When should it be removed? `clean` or `realclean`? @vrurg Can you give some insight?

### CODE REVIEW

File: tools/templates/moar/Makefile.in [hunk 1]
```diff
@@ -68,33 +68,32 @@ R_SETTING_MOAR = RESTRICTED.setting.moarvm
...
   $(R_SETTING_MOAR) \
   @bpm(RUN_RAKUDO_SCRIPT)@ \
-  inst-rakudo-m@moar::obj@ \
-  inst-rakudo-debug-m@moar::obj@ \
-  inst-rakudo@moar::obj@ \
-  inst-rakudo-debug@moar::obj@ \
-  inst-rakudow-m@moar::obj@ \
-  inst-rakudow@moar::obj@ \
-@for_langalias( @bpm(INST_@uclangalias@)@ \
+@for_langalias(  @bpm(INST_@uclangalias@)@ \
```
Comment (MasterDuke17) [inline tools/templates/moar/Makefile.in:71]: Minor nit, but there's inconsistent use of 0, 1, or 2 spaces after an opening `(`.

Comment (patrickbkr) [reply tools/templates/moar/Makefile.in:71]: I that specific spot it's on purpose. All the lines above and below have two leading spaces. If that line stays single space the line is indented one to few in the Makefile.

File: .gitignore [hunk 2]
```diff
@@ -98,26 +98,7 @@ perl6.rc
...
-/inst-rakudo-debug
-/inst-rakudo-m
-/inst-rakudo-debug-m
-/inst-rakudo.exe
-/inst-rakudo-debug.exe
-/inst-rakudo-m.exe
-/inst-rakudo-debug-m.exe
-/inst-rakudow.exe
-/inst-rakudow-m.exe
+/inst-*
```
Comment (vrurg) [inline .gitignore:101]: Some food for thought: if somebody leaves a temporary note for themselves in a _inst-as-something.txt_, for example, then unpleasant surprise may await on them. Or a copy of an _inst-_ file could be made for later analysis/comparison.

I have briefly verified our current approach and found that we only use extension-based cleanup which is much safer with regard to possible unintended damage done.

So, my thinking is that we better consider the source dir as a workplace where a dev might do many kind of things while experimenting with a new feature, a fix, or whatever.

Comment (patrickbkr) [reply .gitignore:101]: I think you missed, that this is the .gitignore and not the cleanup rule in the Makefile. If not, can you clarify what unpleasant surprise you have in mind?

Comment (vrurg) [reply .gitignore:101]: I did miss it, sorry! Ok, then there is no problem here.

**Review by vrurg — APPROVED** [review]

**Pull request merged**